core/src/scene/asset.cpp

```diff
@@ -0,0 +1,135 @@
+#include "scene/asset.h"
+#include "platform.h"
+#include "log.h"
+
+#define MINIZ_HEADER_FILE_ONLY
+#include <miniz.c>
+
+namespace Tangram {
+
+struct ZipHandle {
+    std::unique_ptr<mz_zip_archive> archiveHandle;
+    std::map<std::string, unsigned int> fileIndices = {};
+
+    std::vector<char> data;
+};
+
+Asset::Asset(std::string name, std::string path, std::shared_ptr<ZipHandle> zipHandle,
+        std::vector<char> zippedData) :
+        m_name(name),
+        m_path(path),
+        m_zipHandle(zipHandle) {
+
+    buildZipHandle(zippedData);
+}
+
+Asset::~Asset() {
+    if (m_zipHandle && m_zipHandle->archiveHandle) {
+        mz_zip_reader_end(m_zipHandle->archiveHandle.get());
+    }
+}
+
+bool Asset::isBaseYaml(const std::string& filename) {
+    auto extLoc = filename.find(".yaml");
+    if (extLoc == std::string::npos) { return false; }
+
+    auto slashLoc = filename.find("/");
+    if (slashLoc != std::string::npos) { return false; }
+    return true;
+}
+
+void Asset::buildZipHandle(const std::vector<char>& zipData) {
+
+    if (zipData.empty()) { return; }
+
+    m_zipHandle = std::make_shared<ZipHandle>();
+    m_zipHandle->archiveHandle.reset(new mz_zip_archive());
+    m_zipHandle->data = zipData;
```

Comment:
Now there is a swap. But we need to hold the rawData for the zipbundle for further processing. a81f3d8a79132ad00c645bdbf1edcfb4f91f14fc